tests/test_catalog_compile.py

```diff
@@ -211,6 +211,10 @@ def test_catalog_compile(load_cluster, config: Config, tmp_path: Path, capsys):
     # Stdout success msg
     captured = capsys.readouterr()
     assert "Catalog compiled!" in captured.out
+    assert (
```

Comment:
Can we have a case that tests the deprecation notice is shown?

Reply:
I didn't extend the integration test, since that requires a lot of plumbing to test multiple different configurations (another catalog repo on GitHub etc.), but I added unit tests for `dependency_mgmt._read_components()`.